src/ixdat/calculators/ms_calculators.py

```diff
@@ -0,0 +1,829 @@
+        mass,
+        bg,
+        name=None,
+    ):
+        self.mass = mass
+        self.bg = bg
+        self.name = name or f"{mass} bg"
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(name={self.name}, bg={self.bg})"
+
+    def get_bg_for_t(self, t):
+        return self.bg
+
+
+class MSBackgroundSet(Calculator):
+    calculator_type = "MS background"
+
+    extra_linkers = {
+        "ms_background_constants": ("ms_constant_backgrounds", "ms_constant_bg_ids")
+        # Additional types of backgrounds would go here
+    }
+    child_attrs = [
+        "constant_bg_list",
+        # FIXME: This repeats info in extra_linkers. Should be possible to combine. #75.
+    ]
+
+    def __init__(
+        self,
+        name=None,
+        date=None,
+        bg_list=None,
+        technique="MS",
+        tstamp=None,
+        measurement=None,
+    ):
+        super().__init__(
+            name=name, technique=technique, tstamp=tstamp, measurement=measurement
+        )
+        self.constant_bg_list = []
+        for bg in bg_list:
+            if type(bg) is MSConstantBackground:
+                self.constant_bg_list.append(bg)
+            # elif type(bg) is OtherBackgroundClassWhenItIsReady:
+            #   self.other_bacground_list.append(bg)
+            else:
+                warnings.warn(
+                    f"Initiating an MSBackgoundSet with a background list including {bg}"
+                    ", which is not a recognized MS background type. Skipping that one."
+                )
+
+    @classmethod
+    def from_measurement_point(cls, measurement, tspan, mass_list=None):
+        mass_list = mass_list or measurement.mass_list
+        bg_list = []
+        for mass in mass_list:
+            _, S = measurement.grab(mass, remove_background=False, tspan=tspan)
+            bg_list.append(MSConstantBackground(mass=mass, bg=np.mean(S)))
+        return cls(bg_list=bg_list, measurement=measurement)
+
+    @property
+    def ms_constant_bg_ids(self):
+        return [cal.id for cal in self.constant_bg_list]
+
+    @property
+    def bg_list(self):
+        return self.constant_bg_list  # + self.other_bg_lists_when_they_are_here
+
+    @property
+    def mass_list(self):
+        return list({bg.mass for bg in self.bg_list})
+
+    @property
+    def available_series_names(self):
+        return set([f"{mass}" for mass in self.mass_list])
+
+    def calculate_series(self, mass, measurement=None):
+        """Return a calibrated series for `key` if possible.
+
+        If key starts with "n_", it is interpreted as a molecule flux. This method then
+        searches the calibration for a sensitivity factor for that molecule uses it to
+        divide the relevant mass signal from the measurement. Example acceptable keys:
+        "n_H2", "n_dot_H2".
+        If the key does not start with "n_", or the calibration can't find a relevant
+        sensitivity factor and mass signal, this method returns None.
```

Comment:
This docstring was copy-pasted out of place. I've rewritten it.

This hits a `StopIteration` error rather than returning `None` in the case that the mass is not available.